### ISSUE TITLE
[NEWS] Sandbox - Przerwa Techniczna

### ISSUE THREAD

**Issue by PawelTaberski** [allegro/allegro-api#1999]
**23 lipca w godzinach od 8-00 do 11-00** nastąpi przerwa techniczna na środowisku testowym Allegro.

**W tym czasie wygasimy wszystkie aktywne tokeny** - po zakończeniu przerwy technicznej trzeba będzie się ponownie zautoryzować. Nie trzeba jednak ponownie rejestrować aplikacji. Ponadto REST API i WebAPI Allegro mogą być niedostępne na środowisku testowym.

**Comment by TomekDuszynski** [comment]
Sandbox nawet po utworzeniu nowego powiązania aplikacji z Allegro zwraca 401 przy próbie wykonania jakiegokolwiek requestu.

**Comment by TomekDuszynski** [comment]
Dodam tylko, że przed przerwą wszystko działało jeszcze dziś rano bez zarzutu :)

**Comment by Onixarts** [comment]
Nie można kupować na sandboxie przez www. 
> Wystąpił błąd, spróbuj ponownie.

**Comment by PrzemyslawLukanowski** [comment]
@TomekDuszynski - czy wygenerowałeś nowy token? Błąd wskazuje na to, że posługujesz się niewłaściwym.

**Comment by TomekDuszynski** [comment]
@PrzemyslawLukanowski tak, token generowaliśmy na nowo, a nawet utworzyliśmy nową aplikację i nic.

**Comment by PrzemyslawLukanowski** [comment]
@TomekDuszynski - rozkoduj swój token (wszystkie tokeny są w standardzie JWT) i zweryfikuj, czy należy do usera o przekazanym clientID oraz sprawdź, czy nie minął jego czas ważności. Czy za pomocą refresh tokena jesteś w stanie wygenerować nowy access token? Możesz także przesłać nam przykładowy request i response za pomocą [formularza kontaktowego](https://allegro.pl/pomoc/kontakt) z dopiskiem GitHub #1999.

**Comment by alekskuc** [comment]
Witam, przestały działać tokeny na sandboxie po przerwie technicznej.

```
REQUEST 2019-07-24 10:27:45.1361011
curl -X GET \
  'https://api.allegro.pl.allegrosandbox.pl/after-sales-service-conditions/implied-warranties?seller.id=43955776&limit=1&offset=0' \
  -H 'Accept: application/vnd.allegro.public.v1+json' \
  -H 'Content-Type: application/vnd.allegro.public.v1+json' \
  -H 'Accept-Language: pl-PL' \
  -H 'Authorization: Bearer ' \

RESPONSE 2019-07-24 10:27:45.2110737 Duration: 74,9726ms
{
  "Data": null,
  "Validation": {
    "Errors": null,
    "ValidatedAt": null
  },
  "Exception": null,
  "TraceId": "60ff64eb530f03a3",
  "Error": true,
  "StatusCode": 401,
  "Message": "{\"error\":\"invalid_token\",\"error_description\":\"Cannot convert access token to JSON\"}"
}
```
ID z rozkodowanego tokena:
```
"exp": 1563957683,
"user_name": "43955776"
```
Chyba wrócił stary błąd z 2018? [[REST API] problem z tokenem](https://github.com/allegro/allegro-api/issues/134)

Jeżeli mam wysłać jeszcze cały token z sandboxa (bo traceId nie wystarczy), to proszę o informację co ja mam wybrać w formularzu kontaktowym jak chodzi o błędy z rest api:

![image](https://user-images.githubusercontent.com/8465007/61778447-0b569480-adff-11e9-86d1-b2c018431ab5.png)

Proszę o weryfikacje i poprawę.

**Comment by PrzemyslawLukanowski** [comment]
@alekskuc Wybierz: konto -> rejestracja i aktywacja.

**Comment by alekskuc** [comment]
> @alekskuc Wybierz: konto -> rejestracja i aktywacja.

@PrzemyslawLukanowski Token wysłany.

**Comment by PrzemyslawLukanowski** [comment]
@alekskuc Zweryfikowałem to - wygląda na to, że token, który przesyłasz jest niepełny (informację o exp i userze otrzymasz nawet w takim przypadku). Upewnij się, że przesyłasz prawidłowy. Spróbuj wygenerować nowy token, ten przesłany w zgłoszeniu stracił ważność o 10:41.

**Comment by alekskuc** [comment]
@PrzemyslawLukanowski Tokeny wysyłam jakie mi rest api zwróci, widocznie zwróciło niepełny, proszę o upewnienie, że rest api zwraca zawsze pełne tokeny.

Request był z 10:27 i wtedy token był ważny bo wygasł o 10:41.

Po wygenerowaniu kolejnego tokenu zaczęło działać.

Poobserwuje czy nie pojawią się błędy ponownie.

**Comment by alekskuc** [comment]
> @alekskuc Zweryfikowałem to - wygląda na to, że token, który przesyłasz jest niepełny (informację o exp i userze otrzymasz nawet w takim przypadku). Upewnij się, że przesyłasz prawidłowy. Spróbuj wygenerować nowy token, ten przesłany w zgłoszeniu stracił ważność o 10:41.

@PrzemyslawLukanowski dalej rest api sandbox zwraca tokeny których próba użycia konczy się błędem HTTP401 {\"error\":\"invalid_token\",\"error_description\":\"Cannot convert access token to JSON\"}. 

Tym razem wysłałem przez formularz dodatkowo pełny request z pobrania nowego tokenu przez metodę POST https://allegro.pl.allegrosandbox.pl/auth/oauth/token?grant_type=authorization_code

**Comment by squasher83** [comment]
Potwierdzam:
            [error] => invalid_token
            [error_description] => Cannot convert access token to JSON
W aplikacji nic się nie zmieniło oprócz użycia sandboxa i wygenerowania nowego tokena. Nadmienię, że na serwerze produkcyjnym wszystko działa przy użyciu tej samej aplikacji.

**Comment by TomekDuszynski** [comment]
@PrzemyslawLukanowski przesłałem request i response przez formularz kontaktowy. Sytuacja jest analogiczna jak u poprzedników. Generujemy token przez API ale on nie działa dla żadnego requestu.

**Comment by alekskuc** [comment]
Dzień dobry znalazłem w czym jest problem z tokenami w rest api sandbox. 

W rest api w metodzie https://allegro.pl.allegrosandbox.pl/auth/oauth/token?grant_type=authorization_code została zmieniona długość zwracanych tokenów i teraz przychodzi ponad dwa razy dłuższy "access_token".

Jak użyję refresh token, który jest zwrócony w parze ze zmienionym tokenem, w tej samej metodzie ale z innym grant_type https://allegro.pl/auth/oauth/token?grant_type=refresh_token, zwracany jest "access_token" starej krótszej długości i wszystko zaczyna działać po staremu, tak jak wspominałem wcześniej, że po ponownym pobraniu działa.

@PrzemyslawLukanowski Dlaczego rest api sandbox zwraca teraz około dwa razy dłuższe tokeny w metodzie https://allegro.pl.allegrosandbox.pl/auth/oauth/token?grant_type=authorization_code i nigdzie nie było informacji o takiej zmianie? 

Proszę o przywrócenie poprzedniego działania.

**Comment by squasher83** [comment]
Witam,
Proszę o odpowiedź pracownika Allegro, kiedy zamierzają naprawić problem, bo nie ukrywam, że muszę skończyć projekt w tym miesiącu, a nie mam go gdzie przetestować.
Pozdrawiam

**Comment by PrzemyslawLukanowski** [comment]
Obecnie zwracane tokeny na sandboxie są dłuższe, jednak działają prawidłowo. Przetestowałem odświeżanie tokena, zwracany jest dłuższy access token, który również działa prawidłowo. 
@alekskuc zweryfikowałem token, który przesłałeś - został skrócony o 248 znaków. Upewnij się, czy nie jest to wynikiem działania Twojego oprogramowania.

**Comment by TomekDuszynski** [comment]
Potwierdzam, wydłużone tokeny działają teraz prawidłowo.

**Comment by alekskuc** [comment]
> Obecnie zwracane tokeny na sandboxie są dłuższe, jednak działają prawidłowo. Przetestowałem odświeżanie tokena, zwracany jest dłuższy access token, który również działa prawidłowo.
> @alekskuc zweryfikowałem token, który przesłałeś - został skrócony o 248 znaków. Upewnij się, czy nie jest to wynikiem działania Twojego oprogramowania.

Dzień dobry, oprogramowanie było dostosowane do działania z krótszym tokenem. Wydłużenie ich w rest api spowodowało opisane problemy. 

@PrzemyslawLukanowski Proszę o informowanie z wyprzedzeniem o takich zmianach, oszczędzi to wiele pracy, maili i telefonów wielu osobom integrującym rest api i Klientom korzystających z integracji i naturalnie Wam.

@PrzemyslawLukanowski Jeżeli ta zmiana w długości tokenów musi wejść to dołączam się do pytania @TomekDuszynski kiedy będzie udostępniona w rest api produkcyjnym?

**Comment by PrzemyslawLukanowski** [comment]
Zmiana długości tokena jest przewidziana również na produkcji. Token nie ma zdefiniowanej długości - składa się z nagłówka, body oraz podpisu. Token sie wydluzyl, poniewaz klucz którym jest podpisywany access i refresh token jest jeszcze bardziej bezpieczny.

**Comment by TomekDuszynski** [comment]
@PrzemyslawLukanowski dziękuję za odpowiedź. Brakuje jeszcze bardzo istotnej informacji dla użytkowników API. Kiedy zmiana wejdzie na produkcję?

**Comment by PrzemyslawLukanowski** [comment]
@TomekDuszynski  Na razie konkretny termin nie jest jeszcze znany. Będziemy informować w odpowiednim komunikacie.
